schema.sql

```diff
@@ -5,69 +5,142 @@ drop view if exists unit_attributes;
 
 -- only generation units
 create table  generation_units (
 	unit_id integer primary key,
 	name text not null unique,
-	prime_mover text not null references prime_mover_types(prime_mover),
+	prime_mover text not null,
+	fuel_type text not null,
 	balancing_topology text not null references balancing_topologies(name),
+	start_year integer not null check (start_year >= 0),
 	rating float not null check (rating > 0 ),
 	base_power float not null check (base_power > 0),
-	check (base_power >= rating)
+	check (base_power >= rating),
+	foreign key (prime_mover, fuel_type) references prime_mover_types(prime_mover, fuel_type)
 );
--- ) strict;
--- create storage batteries table
+
 -- only storge units
 create table  storage_units (
 	storage_unit_id integer primary key,
 	name text not null unique,
-	prime_mover text not null references prime_mover_types(prime_mover),
+	prime_mover text not null,
+	fuel_type text not null,
+	max_capacity float not null check (max_capacity > 0),
 	balancing_topology text not null references balancing_topologies(name),
+	charging_efficiency float not null check (charging_efficiency > 0),
+	discharge_efficiency float not null check (discharge_efficiency > 0),
+	start_year integer not null check (start_year >= 0),
 	rating float not null check (rating > 0 ),
 	base_power float not null check (base_power > 0),
-	check (base_power >= rating)
+	check (base_power >= rating),
+	foreign key (prime_mover, fuel_type) references prime_mover_types(prime_mover, fuel_type)
+);
+
+-- create table for technologies, perhaps change the name to investment_technologies
+create table supply_technologies (
+    technology_id integer primary key,
+    prime_mover text not null,
+	fuel_type text not null,
+	technology_class real null,
+    scenario text null,
+	area text null references areas(name),
+	balancing_topology text null references balancing_topologies(name),
+	foreign key (prime_mover, fuel_type) references prime_mover_types(prime_mover, fuel_type)
+);
+
+create table storage_technologies (
+	storage_unit_id integer primary key,
+	name text not null unique,
+	prime_mover text not null,
+	fuel_type text not null,
+	scenario text null,
+	area text null references areas(name),
+	balancing_topology text null references balancing_topologies(name),
+	foreign key (prime_mover, fuel_type) references prime_mover_types(prime_mover, fuel_type)
 );
 
 
+create table operational_data (
+	unit_id references generation_units(unit_id),
+	fom_cost float not null check (fom_cost >= 0),
+	vom_cost float not null check (vom_cost >=0),
+	outage float not null,
+	startup_cost float not null check (startup_cost >= 0),
+	min_stable_level float not null check (min_stable_level >= 0),
+	mttr integer not null check (mttr >= 0),
+	startup_fuel_mmbtu_per_mw float not null check (startup_fuel_mmbtu_per_mw >= 0),
+	plant_uptime float not null check (plant_uptime >= 0),
+	plant_downtime float not null check (plant_downtime >= 0)
```

Comment:
We can simplify this to uptime and downtime

Reply:
Cool done